drivers/rtc/rtc_rv8263.c

```diff
@@ -0,0 +1,918 @@
+        return -EINVAL;
+    }
+
+    k_spinlock_key_t key = k_spin_lock(&data->lock);
+    ret = data->is_alarm_pending ? 1 : 0;
+    data->is_alarm_pending = false;
+    k_spin_unlock(&data->lock, key);
+
+    return ret;
+}
+#endif
+
+#ifdef CONFIG_RTC_UPDATE
+/**
+ * @brief
+ *
+ * @param p_dev         Pointer to device strucure
+ * @param callback      RTC update callback
+ * @param p_user_data   Pointer to user data for update callback
+ * @return int          0 when successful
+ */
+int rv8263_update_callback(const struct device *p_dev, rtc_update_callback callback, void *p_user_data)
+{
+    struct rv8263c8_data *const data = p_dev->data;
+
+    if (p_dev == NULL) {
+        return -EINVAL;
+    }
+
+    k_spinlock_key_t key = k_spin_lock(&data->lock);
+    data->update_cb = callback;
+    data->update_cb_data = p_user_data;
+    k_spin_unlock(&data->lock, key);
+
+    return rv8263c8_update_enable_timer(p_dev);
+}
+#endif
+
+#ifdef CONFIG_RTC_CALIBRATION
+/**
+ * @brief
+ *
+ * @param p_dev         Pointer to device strucure
+ * @param calibration
+ * @return int          0 when successful
+ */
+int rv8263c8_calibration_set(const struct device *p_dev, int32_t calibration)
+{
+    int err;
+    int8_t reg;
+    struct rv8263c8_data *const data = p_dev->data;
+    const struct rv8263c8_config *config = p_dev->config;
+
+    if (p_dev == NULL) {
+        return -EINVAL;
+    }
+
+    reg = calibration & 0x7F;
+    reg = bin2bcd(reg);
+
+    k_spinlock_key_t key = k_spin_lock(&data->lock);
+    err = i2c_reg_update_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_OFFSET, 0x7F, reg);
+    k_spin_unlock(&data->lock, key);
+
+    return err;
+}
+
+/**
+ * @brief
+ *
+ * @param p_dev         Pointer to device strucure
+ * @param p_calibration
+ * @return int          0 when successful
+ */
+int rv8263c8_calibration_get(const struct device *p_dev, int32_t *p_calibration)
+{
+    int err;
+    uint8_t value;
+    struct rv8263c8_data *const data = p_dev->data;
+    const struct rv8263c8_config *config = p_dev->config;
+
+    if ((p_dev == NULL) || (p_calibration ==  NULL)) {
+        return -EINVAL;
+    }
+
+    k_spinlock_key_t key = k_spin_lock(&data->lock);
+    err = i2c_reg_read_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_OFFSET, &value);
+    k_spin_unlock(&data->lock, key);
+    if (err < 0) {
+        LOG_ERR("Error while reading OFFSET! Error: %i", err);
+        return err;
+    }
+
+    *p_calibration = bcd2bin(value & 0x7F);
+
+    return 0;
+}
+#endif
+
+static const struct rtc_driver_api rv8263c8_driver_api = {
+    .set_time = rv8263c8_time_set,
+    .get_time = rv8263c8_time_get,
+#ifdef CONFIG_RTC_ALARM
+    .alarm_get_supported_fields = rv8263c8_alarm_get_supported_fields,
+    .alarm_set_time = rv8263c8_alarm_set_time,
+    .alarm_get_time = rv8263c8_alarm_get_time,
+    .alarm_is_pending = rv8263c8_alarm_is_pending,
+    .alarm_set_callback = rv8263c8_alarm_set_callback,
+#endif
+#ifdef CONFIG_RTC_UPDATE
```

Comment:
This should be updated to
```
#if CONFIG_RTC_UPDATE && DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)

#endif
```

furthermore the API itself must check if the int_gpios actually exists for that node, which can be done by checking `data->int_gpio->port != NULL`